src/sass/toggle-button-group/toggle-button-group.scss

```diff
@@ -89,23 +143,199 @@
     max-width: initial;
 }
 
-/* Columns */
-.toggle-button-group[data-columns="2"] li {
-    width: calc(100% / 2 - 11px);
+/* Responsive Layouts */
+// first, we define media queries as a fallback for browsers not supporting container queries
+@supports not (contain: inline-size) {
+    @media (min-width: $_screen-size-XS) {
+        // Responsive columns based on XS breakpoint per button type
+        .toggle-button-group ul {
+            grid-template-columns: repeat(
+                var(--row-min-toggle-buttons-xs),
+                1fr
+            );
+        }
+
+        .toggle-button-group--list-layout ul {
+            grid-template-columns: repeat(
+                var(--row-list-toggle-buttons-xs),
+                1fr
+            );
+        }
+
+        .toggle-button-group--gallery-layout ul {
+            grid-template-columns: repeat(
+                var(--row-gallery-toggle-buttons-xs),
+                1fr
+            );
+        }
+
+        // Explicit developer overrides for XS breakpoint for exception cases
+        @include toggle-button-grid-templates(xs, 16);
+    }
+
+    @media (min-width: $_screen-size-SM) {
+        // Responsive columns based on SM breakpoint per button type
+        .toggle-button-group ul {
+            grid-template-columns: repeat(
+                var(--row-min-toggle-buttons-sm),
+                1fr
+            );
+        }
+
+        .toggle-button-group--list-layout ul {
+            grid-template-columns: repeat(
+                var(--row-list-toggle-buttons-sm),
+                1fr
+            );
+        }
+
+        .toggle-button-group--gallery-layout ul {
+            grid-template-columns: repeat(
+                var(--row-gallery-toggle-buttons-sm),
+                1fr
+            );
+        }
+
+        // Explicit developer overrides for SM breakpoint for exception cases
+        @include toggle-button-grid-templates(sm, 16);
+    }
+
+    @media (min-width: $_screen-size-MD) {
+        // Responsive columns based on MD breakpoint per button type
+        .toggle-button-group ul {
+            grid-template-columns: repeat(
+                var(--row-min-toggle-buttons-md),
+                1fr
+            );
+        }
+
+        .toggle-button-group--list-layout ul {
+            grid-template-columns: repeat(
+                var(--row-list-toggle-buttons-md),
+                1fr
+            );
+        }
+
+        .toggle-button-group--gallery-layout ul {
+            grid-template-columns: repeat(
+                var(--row-gallery-toggle-buttons-md),
+                1fr
+            );
```

Comment:
Im wondering, why do we have this together with the mixins?

Reply:
These are the internal `toggle-button-group` rules for prescribing how many buttons should be on each line before wrapping, and they're established by us. On the other hand, the classes generated by the mixin allow for the uses of the exception cases where the dev can use custom rules for specific cases. It's `#5` in the PR description.